app/structure/every-fucking-thing-factory.js

```diff
@@ -55,31 +57,35 @@ export default class EveryFuckingThingFactory {
     }
 
     createPlayers() {
-        return new PlayerCollection([
-            new Player(0, "andrzej", this.createAlgorithmForPlayer(0), this.createSnakeForPlayer(0)),
-            new Player(1, "maciek", this.createAlgorithmForPlayer(1), this.createSnakeForPlayer(1)),
-            new Player(2, "mwl", this.createAlgorithmForPlayer(2), this.createSnakeForPlayer(2))
-        ]);
-    }
-
+        const players = [];
+        const algorithms = AlgorithmLoader.load(SNAKES_DIR, INDEX_FILE);
+        const maxSnakes = INITIAL_SNAKES_DATA.length;
 
-    getPlayersSnakes(players) {
-        return players.getSnakes();
-    }
+        if (algorithms.length > maxSnakes) {
+            throw new Error(`I refuse to let more than ${maxSnakes} snakes to the game! Don't ask me why ;-(`);
+        }
 
-    createSnakeForPlayer(index) {
-        return this.createSnakeFromConfiguration(index, this.getSnakeConfigForPlayer(index));
-    }
+        algorithms.forEach(algorithm => {
+            const AlgorithmConstructor = algorithm.constructor;
+            const name = algorithm.name;
+            const index = players.length;
+            const initialSnakeData = this.getInitialSnakeData(index);
+
+            players.push(new Player(
```

Comment:
Index data is used in application please don't remove it.

Reply:
Where? Can't find it.

If so, for the sake of code readability it would be better if it was accessed by getter. Without that one may think that it was left here by accident.
There are some people who state that there is nothing wrong with accessing simple public variables directly (esp. inside package), but their argument ignores issues of "code discoverablity".

Also, index seems to be pretty bad name for it. It suggest that it has to do something with order and order is oc defined by collection which keeps Player object. Maybe it would be better to use "name" instead? Or rename it "id"?

Enough said, we shouldn't remove/refactor code we don't fully understand and it's only my "evil me" who came to play in this case. 😈